### PR TITLE
Dilpreet |Test | UI - Currect the GSTR 1 and GSTR 2 Design.

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/2nau7vb

### CODE REVIEW

**Comment by dilpreetsinghofficial** [comment]
Reviewed

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts [hunk 1]
```diff
@@ -56,6 +58,7 @@ export class FilingHeaderComponent implements OnInit, OnChanges, OnDestroy {
     /** True if current organization is company */
     @Input() public isCompany: boolean;
     @ViewChild('cancelConfirmationModel', { static: true }) public cancelConfirmationModel: ModalDirective;
+    @ViewChild('pushToPortalModel', { static: true }) public pushToPortalModel: ModalDirective;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts:62]: add JS doc

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts [hunk 2]
```diff
@@ -248,4 +261,57 @@ export class FilingHeaderComponent implements OnInit, OnChanges, OnDestroy {
...
         this.cancelConfirmationModel.toggle();
     }
+
+    /**
+     * This will use for period change
+     *
+     * @param {*} ev
+     * @memberof FilingHeaderComponent
+     */
+    public periodChanged(ev): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts]:
```suggestion
    public periodChanged(event?: any): void {
```

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts [hunk 3]
```diff
@@ -248,4 +261,57 @@ export class FilingHeaderComponent implements OnInit, OnChanges, OnDestroy {
...
+    * @memberof FilingHeaderComponent
+    */
+    public onDownloadJsonGSTR(type: string): void {
+        if (this.activeCompanyGstNumber) {
+            let request: GstrJsonDownloadRequest = new GstrJsonDownloadRequest();
+            request.type = type;
+            request.gstin = this.activeCompanyGstNumber;
+            request.from = this.currentPeriod.from;
+            request.to = this.currentPeriod.to;
+            this.gstReconcileService.downloadGSTRJSON(request).subscribe(res => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts]:
```suggestion
            this.gstReconcileService.downloadGSTRJSON(request).pipe(takeUntil(this.destroyed$)).subscribe(res => {
```

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts [hunk 4]
```diff
@@ -248,4 +261,57 @@ export class FilingHeaderComponent implements OnInit, OnChanges, OnDestroy {
...
+                if (res.status === "success") {
+                    let blobData = this.generalService.base64ToBlob(res?.body, "json", 512);
+                    return saveAs(blobData, `${this.activeCompanyGstNumber}.json`);
+                } else {
+                    this.toasty.errorToast(res?.body.message);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.ts]: are you getting error message inside body?

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                    <p>{{ localeData?.filing?.excel_file_support }}</p>
+                </li>
+                <li class="list-group-item">
+                    <p>{{ localeData?.filing?.organised_accountant }}</p>
+                </li>
+            </ul>
+            <a
+                class="btn btn-sky-blue mr-0 downloads-button json-button"
+                target="_blank"
+                href="https://www.gst.gov.in/download/returns "
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.html]:
```suggestion
                href="https://www.gst.gov.in/download/returns"
```

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.left-side-center {
+    text-align: center;
+}
+.left-side {
+    border-right: 1px solid #272727;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.scss]: don't hardcode color

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+.left-side-center {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.scss]: use scss pattern

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import { Component, EventEmitter, Input, Output, OnInit } from '@angular/core';
+@Component({
+    selector: 'push-to-portal',
+    templateUrl: './push-to-portal.component.html',
+    styleUrls: ['./push-to-portal.component.scss']
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts]: add changedetection strategy

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+    constructor() {
+    }
+
+    /**
+     * Initializes the component
+     *
+     * @memberof PushtToPortalComponent
+     */
+    public ngOnInit(): void {}
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts]: what's the use of blank methods?

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * This will use for download success sheet
+     *
+     * @param {Event} e
+     * @memberof PushtToPortalComponent
+     */
+    public onDownloadExcel(e: Event): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts]:
```suggestion
    public onDownloadExcel(event: Event): void {
```

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        this.downloadExcel.emit(e);
+    }
+
+    /**
+ * This will use for download success sheet
+ *
+ * @param {Event} e
+ * @memberof PushtToPortalComponent
+ */
+    public onDownloadJson(e: Event): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts]:
```suggestion
    public onDownloadJson(event: Event): void {
```

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+     *
+     * @param {Event} e
+     * @memberof PushtToPortalComponent
+     */
+    public onDownloadExcel(e: Event): void {
+        this.downloadExcel.emit(e);
+    }
+
+    /**
+ * This will use for download success sheet
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts:29]: format code

File: apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        this.downloadJson.emit(e);
+    }
+
+    /**
+     *This will emit the cancel popup  hide
+     *
+     * @param {Event} e
+     * @memberof PushtToPortalComponent
+     */
+    public onCancel(e: Event): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/modals/push-to-portal/push-to-portal.component.ts]:
```suggestion
    public onCancel(event: Event): void {
```

File: apps/web-giddh/src/app/services/GstReconcile.service.ts [hunk 14]
```diff
@@ -246,4 +246,25 @@ export class GstReconcileService {
             .replace(':companyUniqueName', this.companyUniqueName))
             .pipe(catchError((error) => this.errorHandler.HandleCatch<any, any>(error)));
     }
+
+    /**
+ * This will use for download json for gstr1
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/GstReconcile.service.ts]: format code

File: apps/web-giddh/src/assets/locale/gst/en.json [hunk 15]
```diff
@@ -108,7 +108,15 @@
...
         "cancel_title": "Are you sure you want to cancel it?",
-        "total_invoice_value": "Total Invoice Value"
+        "total_invoice_value": "Total Invoice Value",
+        "errors":"Errors",
+        "push_to_portal":"Push to Portal",
+        "gst_offline_utility":"GST Offline Utility",
+        "download_excel":" Download Excel",
+        "excel_file_support":"1. This Excel file support , GSTIN Portal Offline tool.",
+        "organised_accountant":" 2. Its an organised way to give to any accountant/CA.",
+        "json_support":" JSON file support by GST Portal",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/en.json]:
```suggestion
        "json_support":"JSON file support by GST Portal",
```

File: apps/web-giddh/src/assets/locale/gst/en.json [hunk 16]
```diff
@@ -108,7 +108,15 @@
...
+        "download_json":" Download JSON"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/en.json]:
```suggestion
        "download_json":"Download JSON"
```

File: apps/web-giddh/src/assets/locale/gst/en.json [hunk 17]
```diff
@@ -108,7 +108,15 @@
         "transaction_date": "Transaction Date",
         "reconcilation_note": "All transactions in Giddh Books whose vendor GSTIN, invoice number, invoice date, invoice amount and tax amount are same as the transactions in GSTN are displayed in this section.",
         "cancel_title": "Are you sure you want to cancel it?",
-        "total_invoice_value": "Total Invoice Value"
+        "total_invoice_value": "Total Invoice Value",
+        "errors":"Errors",
+        "push_to_portal":"Push to Portal",
+        "gst_offline_utility":"GST Offline Utility",
+        "download_excel":" Download Excel",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/en.json]:
```suggestion
        "download_excel":"Download Excel",
```

File: apps/web-giddh/src/assets/locale/gst/hi.json [hunk 18]
```diff
@@ -108,7 +108,15 @@
 		"transaction_date": "कार्यवाही की तिथि",
 		"reconcilation_note": "गिद्ध बुक्स में सभी लेनदेन जिनके विक्रेता जीएसटीआईएन, चालान संख्या, चालान तिथि, चालान राशि और कर राशि जीएसटीएन में लेनदेन के समान हैं, इस खंड में प्रदर्शित होते हैं।",
 		"cancel_title": "क्या आप वाकई इसे रद्द करना चाहते हैं?",
-        "total_invoice_value": "कुल चालान मूल्य"
+        "total_invoice_value": "कुल चालान मूल्य",
+        "errors":"त्रुटियों",
+        "push_to_portal":"पोर्टल पर पुश करें",
+        "gst_offline_utility":"जीएसटी ऑफ़लाइन उपयोगिता",
+        "download_excel":" एक्सेल डाउनलोड करें",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/hi.json]:
```suggestion
        "download_excel":"एक्सेल डाउनलोड करें",
```

File: apps/web-giddh/src/assets/locale/gst/hi.json [hunk 19]
```diff
@@ -108,7 +108,15 @@
...
+        "excel_file_support":"1. यह एक्सेल फाइल सपोर्ट, जीएसटीआईएन पोर्टल ऑफलाइन टूल।",
+        "organised_accountant":"2. यह किसी भी एकाउंटेंट/सीए को देने का एक संगठित तरीका है।",
+        "json_support":"GST पोर्टल द्वारा JSON फ़ाइल का समर्थन",
+        "download_json":" JSON डाउनलोड  करें"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/hi.json]:
```suggestion
        "download_json":"JSON डाउनलोड  करें"
```

File: apps/web-giddh/src/assets/locale/gst/mr.json [hunk 20]
```diff
@@ -108,7 +108,15 @@
 		"transaction_date": "व्यवहाराची तारीख",
 		"reconcilation_note": "जीएसडीएन मधील व्यवहार जीएसटीएन, चलन क्रमांक, बीजक तारीख, चलन रक्कम आणि कराची रक्कम याप्रमाणेच गिद्ध पुस्तकांमधील सर्व व्यवहार या विभागात दर्शविलेले आहेत.",
 		"cancel_title": "आपली खात्री आहे की आपण ते रद्द करू इच्छिता?",
-        "total_invoice_value": "एकूण चलन मूल्य"
+        "total_invoice_value": "एकूण चलन मूल्य",
+        "errors":"चुका",
+        "push_to_portal":"पोर्टलवर पुश करा",
+        "gst_offline_utility":"जीएसटी ऑफलाइन उपयुक्तता",
+        "download_excel":" एक्सेल डाउनलोड करा",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/mr.json]:
```suggestion
        "download_excel":"एक्सेल डाउनलोड करा",
```

File: apps/web-giddh/src/assets/locale/gst/mr.json [hunk 21]
```diff
@@ -108,7 +108,15 @@
...
+        "excel_file_support":"1. हे एक्सेल फाइल सपोर्ट, जीएसटीआयएन पोर्टल ऑफलाइन टूल.",
+        "organised_accountant":"2. कोणत्याही अकाउंटंट/सीएला देण्याचा हा एक संघटित मार्ग आहे.",
+        "json_support":" GST पोर्टलद्वारे JSON फाइल समर्थन",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/mr.json]:
```suggestion
        "json_support":"जीएसटी पोर्टलद्वारे JSON फाइल समर्थन",
```

File: apps/web-giddh/src/assets/locale/gst/mr.json [hunk 22]
```diff
@@ -108,7 +108,15 @@
...
+        "download_json":" JSON डाउनलोड करा"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/gst/mr.json]:
```suggestion
        "download_json":"JSON डाउनलोड करा"
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**